pageserver/src/waldecoder.rs

```diff
@@ -408,15 +425,12 @@ impl XlXactParsedRecord {
     /// Decode a XLOG_XACT_COMMIT/ABORT/COMMIT_PREPARED/ABORT_PREPARED
     /// record. This should agree with the ParseCommitRecord and ParseAbortRecord
     /// functions in PostgreSQL (in src/backend/access/rmgr/xactdesc.c)
-    pub fn decode(decoded: &DecodedWALRecord) -> XlXactParsedRecord {
-        let info = decoded.xl_info & pg_constants::XLOG_XACT_OPMASK;
-        let mut buf = decoded.record.clone();
-        buf.advance(decoded.main_data_offset);
-
+    pub fn decode(buf: &mut Bytes, mut xid: TransactionId, xl_info: u8) -> XlXactParsedRecord {
+        let info = xl_info & pg_constants::XLOG_XACT_OPMASK;
         // The record starts with time of commit/abort
-        let xact_time = buf.get_i64_le();
+        let xact_time = buf.get_u64_le();
```

Comment:
xact_time is TimestampTz in Postgres, which is signed.

Reply:
Yes, I know it.
But in xlog_utils.rs it is defined as u64.
It doesn't cause any problems now because comparison of time is never used in zenith.
But may be we should fix it to prevent possible problems in future when such comparison will be used.
Should I fix it in this branch or create separate PR for it?

Reply:
Separate PR seems appropriate.

Reply:
Done
https://github.com/zenithdb/zenith/pull/298